src/engine/world/world.cpp

```diff
@@ -1061,7 +1061,9 @@ void World::load(const std::optional<size_t>& slot)
   getPresenter().disableScreenOverlay();
 }
 
-void World::save(const std::filesystem::path& filename, bool isQuicksave)
+void World::save(const std::filesystem::path& filename,
+                 const std::optional<std::filesystem::path> metaPath,
+                 bool isQuicksave)
```

Comment:
It's not really necessary to have a separate parameter for the meta filepath. It's sufficient to just replace the file extension of `filename` with `.meta.yaml` here and in all other places the meta filepath is required.